virtual-desktop/src/app/window-manager/mvd-window-manager/shared/window-manager.service.ts

```diff
@@ -208,6 +219,44 @@ export class WindowManagerService implements MVDWindowManagement.WindowManagerSe
     return parentViewportElement.getAttribute(viewportIdAttr);
   }
 
+  switchWindow(zDistance:number): void {
+    let windows:DesktopWindow[] = this.getAllWindows();
+  
+    if(this.focusedWindow != null) {
+      const focusedWindowId :number = this.focusedWindow.windowId; 
+      windows = windows.filter( (val: DesktopWindow ) => 
+        val.windowId !== focusedWindowId
+      );
+    }  
+
+    if(windows.length>0) {
+      const sortWindows: DesktopWindow[] = windows.sort(
+                          (val1:DesktopWindow, val2:DesktopWindow) => 
+                          ((val1.windowState.zIndex - val2.windowState.zIndex) 
+                          * zDistance)
+                        );
+
+      const windowIds = sortWindows.map((val:DesktopWindow) => val.windowId);
+
+      if(windowIds.length>0) {
+        const selectIdx: number = (Math.abs(zDistance) -1) % (windows.length);
+        const windowId = windowIds[selectIdx];
+        if(this.focusedWindow != null && zDistance<1) {
+          const replaceZIndex = Math.abs(sortWindows[windowIds.length-1].windowState.zIndex)-1;
+          if(replaceZIndex>0) {
+            this.focusedWindow.windowState.zIndex=replaceZIndex;
+          } else {
+            sortWindows.forEach((w,i)=>{
```

Comment:
reseting zValue to higher value, so they don't go negative and disappear off screen